vshard/cfg.lua

```diff
@@ -130,6 +147,9 @@ local function check_replicas(replicas)
     local ctx = {master = false}
     for _, replica in pairs(replicas) do
         validate_config(replica, replica_template, ctx)
+        if replica.listen == nil then
+            replica.listen = replica.uri
+        end
```

Comment:
It looks a bit tricky. "Check" methods supposed to be read-only I think.

Reply:
They fill default fields now. But defaults work only for constant values. Here I tried to make default of one field equal to another field. Anyway, I dropped this in the new version - the tests on old versions were failing because they don't support that option, and it is not needed much anyway.